src/cargo/ops/vendor.rs

```diff
@@ -208,8 +211,17 @@ fn sync(
         };
 
         sources.insert(id.source_id());
-        let dst = canonical_destination.join(&dst_name);
-        to_remove.remove(&dst);
+
+        let dst = if contains_duplicate_versions {
```

Comment:
This path is constructed in a duplicate location down below, so could this logic be either extracted to a helper function/closure or a separate map built up once? (to share the implementation and ensure the logic for calculating the destination is always the same)